.github/workflows/CI.yml

```diff
@@ -12,15 +15,26 @@ jobs:
       fail-fast: false
       matrix:
         version:
-          # Test on the LTS, Invenia prod release and latest release
-          - 1.0.5
-          - 1.3
           - 1
         os:
           - ubuntu-latest
+          - macOS-latest
+          - windows-latest
         arch:
           - x64
-          # - x86
+        include:
+          # Add a 1.0 job just to make sure we still support it
+          - os: ubuntu-latest
+            version: 1.0.5
+            arch: x64
+          # Add a 1.3 job because that's what Invenia actually uses
+          - os: ubuntu-latest
+            version: 1.3
+            arch: x64
+          # Add a 32-bit job to ensure we don't have any 64-bit specific logic
+          - os: ubuntu-latest
+            version: 1
+            arch: x86
```

Comment:
If we're gonna test on x86 we might want to do that for all julia versions in case there are memory issues.